app/src/org/commcare/preferences/FilePreferenceDialogFragmentCompat.java

```diff
@@ -60,17 +61,21 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent) {
             if (resultCode == RESULT_OK && intent != null) {
                 Uri uri = intent.getData();
                 if (uri != null) {
-                    String fileName = FileUtil.getFileName(getContext(), uri);
-                    File destination = new File(getContext().getExternalCacheDir(), fileName);
                     try {
+                        String fileName = FileUtil.getFileName(getContext(), uri);
+                        File destination = new File(getContext().getExternalCacheDir(), fileName);
                         InputStream inputStream = getContext().getContentResolver().openInputStream(uri);
                         FileUtil.copyFile(inputStream, destination);
+                        validateFile(destination.getAbsolutePath());
+                    } catch (FileExtensionNotFoundException e) {
+                        Logger.exception(LogTypes.TYPE_ERROR_STORAGE, e);
```

Comment:
Same comment as above. These two catch paths are essentially the same as long as a FileExtensionNotFoundException  becomes an IOException